master/buildbot/reporters/gerrit_verify_status.py

```diff
@@ -0,0 +1,198 @@
+        payload = {
+            'name': name,
+            'value': value}
+
+        if abstain is not None:
+            payload['abstain'] = abstain
+
+        if rerun is not None:
+            payload['rerun'] = rerun
+
+        if comment is not None:
+            payload['comment'] = comment
+
+        if url is not None:
+            payload['url'] = url
+
+        if reporter is not None:
+            payload['reporter'] = reporter
+
+        if category is not None:
+            payload['category'] = category
+
+        if duration is not None:
+            payload['duration'] = duration
+
+        if self.verbose:
+            log.info(
+                'Sending Gerrit status for {change_id}/{revision_id}: data={data}',
+                change_id=change_id, revision_id=revision_id, data=payload)
+
+        return self._http.post('/'.join([
+            'changes', str(change_id), 'revisions', str(revision_id), 'verify-status~verifications']),
+            json=payload)
+
+    def formatDuration(self, duration):
+        """Format the duration.
+
+        This method could be overriden if really needed, as the duration format in gerrit
+        is an arbitrary string.
+        :param duration: duration in timedelta
+        """
+        days = duration.days
+        hours, remainder = divmod(duration.seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
+        if days:
+            return '{} day{} {}h {}m {}s'.format(
+                days, "s" if days > 1 else "",
+                hours, minutes, seconds)
+        elif hours:
+            return '{}h {}m {}s'.format(hours, minutes, seconds)
+        else:
+            return '{}m {}s'.format(minutes, seconds)
+
+    def getGerritChanges(self, props):
+        """ Get the gerrit changes
+
+            Must return a list of dictionary with at list change_id, and revision_id,
+            which format is the one accepted by the gerrit REST API as of
+            /changes/:change_id/revision/:revision_id paths (see gerrit doc)
+            :param props: an IProperty
+        """
+        if 'gerrit_changes' in props:
+            return props.getProperty('gerrit_changes')
+
+        if 'event.change.number' in props:
+            return [{
+                'change_id': props.getProperty('event.change.number'),
+                'revision_id': props.getProperty('event.patchSet.number')}
+            ]
+        return []
+
+    @defer.inlineCallbacks
+    def send(self, build):
+        props = Properties.fromDict(build['properties'])
+        if build['complete']:
+            value = self.RESULTS_TABLE.get(build['results'], self.DEFAULT_RESULT)
+            comment = yield props.render(self.endDescription)
+            duration = self.formatDuration(build['complete_at'] - build['started_at'])
+
+        else:
+            value = 0
+            comment = yield props.render(self.startDescription)
+            duration = 'pending'
+
+        print build['results'], value
```

Comment:
i think logging should be used here (if it's not a debug print, in the latter case -- remove)